sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -818,6 +818,7 @@ primaryExpression
 
 constant
     : NULL                                                                                     #nullLiteral
+    | INTERVAL STRING AS? identifier?                                                          #maybeAliasedIntervalLiteral
```

Comment:
can you explain how the bug triggers?

Reply:
OK, I will update pr description after ci passes

Reply:
Before this PR, these cases will be captured by grammar rule `multiUnitsInterval` but not `typedLiteral`, e.g. for interval '1 day' day, the value is `1 day` and the unit is day， which actually should be parsed as `((interval 1 day) as day)`

In this PR, I add a new rule "INTERVAL STRING AS? identifier? " to handle interval typed literals and 1-value-unit interval together and before the rule `multiUnitsInterval`.

when the "AS" key is not null, the `STRING` must be the whole context of the interval whether it is valid or not, the identifier should be the alias name or else the "AS" itself as alias name.

when the "AS" key is null
    1 and when the `STRING` contains any alphabet, we can assume that the  `STRING` is the whole context of the interval whether it is valid or not. the identifier part will be alias name if present
    2 and when the `STRING` contains no alphabet, e.g. '1' day, '-1' day, '\t1' day, so the identifier here should be the unit part whether it is valid or not.

Also updated in PR desc.

Reply:
can you check `interval '1 day' day` in other systems? To see if they also treat '1 day' as the number of days and fail.

Reply:
```
postgres=# select interval '1 day' day;
 interval
----------
 1 day
(1 row)

postgres=# select interval '1' day day;
ERROR:  syntax error at or near "day"
LINE 1: select interval '1' day day;
                                ^
postgres=# select interval '1' day;
 interval
----------
 1 day
(1 row)
```

Reply:
```sql
presto> select interval '1 day' day;
Query 20200903_015817_00008_gec8v failed: Invalid INTERVAL DAY value: 1 day

presto> select interval '1' day day;
      day
----------------
 1 00:00:00.000
(1 row)

Query 20200903_015821_00009_gec8v, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]

presto> select interval '1' day;
     _col0
----------------
 1 00:00:00.000
(1 row)

Query 20200903_015825_00010_gec8v, FINISHED, 1 node
Splits: 17 total, 17 done (100.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]

presto>
```

Reply:
presto does not support typed literal for intervals